lib/skywalking/tracer.lua

```diff
@@ -25,22 +25,27 @@ local json = require('cjson.safe')
 local metadata_shdict = ngx.shared.tracing_buffer
 local ngx = ngx
 local nginxComponentId = 6000
-
-
 local Tracer = {}
 
 
 function Tracer:start(upstream_name, correlation)
     local serviceName = metadata_shdict:get("serviceName")
     local serviceInstanceName = metadata_shdict:get('serviceInstanceName')
+    local req_uri = ngx.var.uri
+    if serviceInstanceName == nil or serviceName == nil or (Util.checkIgnoreSuffix(req_uri)) then
+        local tracingContext =  TC.newNoOP()
```

Comment:
If you want to switch to `no_op`, please print a warning in the log, because in `==nil` case, the plugin should not be activated.

Reply:
ok.

Reply:
> If you want to switch to `no_op`, please print a warning in the log, because in `==nil` case, the plugin should not be activated.

every ignored request print a warning，Will it be too much?

Reply:
I am not saying `ignored` trace. You are adding logic here `serviceInstanceName == nil or serviceName == nil`. I mean this is a critical error.

Reply:
> I am not saying `ignored` trace. You are adding logic here `serviceInstanceName == nil or serviceName == nil`. I mean this is a critical error.
got.

Reply:
@alonelaval Don't forget this.

Reply:
> @alonelaval Don't forget this.

ok